server/db/db.js

```diff
@@ -146,34 +132,29 @@ var visit = {
     var query = 'UPDATE visits SET visitedAt = $1, updatedAt = $2, url = $3, ' +
                 'urlHash = $4, rawUrl = $5, title = $6 ' +
                 'WHERE fxaId = $7 AND id = $8 RETURNING id, fxaId, visitedAt, url, urlHash, title';
-    pg.connect(dbParams, function(err, client, done) {
-      if (err) { return onConnectionError(err, cb); }
-      client.query(query, [visitedAt, new Date().toJSON(), url, urlHash, url, title, fxaId, visitId],
-        function(err, r) {
-        if (err) {
-          log.warn('error updating visit: ' + err);
-        } else {
-          _verbose('db.visit.update succeeded');
-        }
-        done();
-        cb(err, r && visit._normalize(r.rows[0]));
-      });
+    var args = [visitedAt, new Date().toJSON(), url, urlHash, url, title, fxaId, visitId];
+
+    db.query(query, args, function(err, r) {
+      if (err) {
+        log.warn('error updating visit: ' + err);
+      } else {
+        _verbose('db.visit.update succeeded');
+      }
+      cb(err, r && visit._normalize(r[0]));
```

Comment:
So, we may be able to easily switch this to using Promises using something like this:

```js
var queryp = Promise.denodeify(db.query);
// ...
return queryp(query, args).then(function (rows) {
  return visit._normalize(rows[0]);
}, function (err) {
  log.warn('error updating visit: ' + err);
  return err;
});
```

Then a bit of server route tweaking to convert from callbacks to promises...
```js
var onError(err) {
  log.warn(err);
  reply(Boom.create(500));
};
visits.get(fxaId, request.query.count).then(reply, onError);
```